### PR TITLE
feat: surface ReservationPath and Reservation

### PR DESCRIPTION
**Pull request by anguillanneuf**
- Surface Reservation and ReservationPath
- Reorder types

### CODE REVIEW

File: google/cloud/pubsublite/types/__init__.py [hunk 1]
```diff
@@ -13,23 +13,26 @@
 # limitations under the License.
 #
 
+from pubsublite_v1.types.common import Reservation
```
Comment (tseaver) [inline google/cloud/pubsublite/types/__init__.py:16]:
```suggestion
from google.cloud.pubsublite_v1.types.common import Reservation
```

**Review by tseaver — CHANGES_REQUESTED** [review]

**Comment by dpcollins-google** [comment]
Authorizing CLA: not sure why the bot doesn't like tianzi :/

**Review by dpcollins-google — APPROVED** [review]

File: google/cloud/pubsublite/types/__init__.py [hunk 2]
```diff
@@ -13,23 +13,26 @@
 # limitations under the License.
 #
 
+from google.cloud.pubsublite_v1.types.common import Reservation
```
Comment (dpcollins-google) [inline google/cloud/pubsublite/types/__init__.py:16]: since "Topic" and "Subscription" aren't here, I don't think "Reservation" should be either?

**Review by dpcollins-google — CHANGES_REQUESTED** [review]
One comment

File: google/cloud/pubsublite/types/__init__.py (hunk 2 above)
Comment (anguillanneuf) [reply google/cloud/pubsublite/types/__init__.py]: that's right. Removed.

**Review by dpcollins-google — APPROVED** [review]

**Review by tseaver — APPROVED** [review]

**Comment by tseaver** [comment]
@googlebot I consent.

**Pull request merged**